### DISCUSSION TITLE
Emoji in offline environment

### DISCUSSION THREAD

**Discussion by earthwalker31** [squidfunk/mkdocs-material#5965] [Q&A]
When hosting the wiki in a network without internet connection, emojis dont seem to render correctly, because there is no connection to: https://cdn.jsdelivr.net/gh/jdecked/twemoji@14.1.2/assets/svg/XXXXX.svg 

![mkdocs](https://github.com/squidfunk/mkdocs-material/assets/56949417/d885d998-c78b-462e-a967-ef2cf7750eec)

This is my config: 

```
markdown_extensions:
  - pymdownx.emoji:
  	emoji_index: !!python/name:materialx.emoji.twemoji
  	emoji_generator: !!python/name:materialx.emoji.to_svg
```
Is there any way of self hosting the twemoji converter? Are the other emoji indexes available offline?

**Comment by squidfunk** [comment]
Excellent question! This problem is solved by the [privacy plugin](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#built-in-privacy-plugin), which downloads all external resources (emojis, images, style sheets, scripts, etc.) during `mkdocs build` and bundles them with your documentation, so they can be viewed without a connection.

You only need to add one line of config to the `plugins` setting in `mkdocs.yml`.

``` yaml
plugins:
  - privacy
```

Note that the privacy plugin is currently only available in [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/), and thus reserved to [sponsors](http://github.com/sponsors/squidfunk) ☺️ Sponsorships start at $15 for __non-commercial__ use and at $125 for __commercial use__.

**Comment by kamilkrzyskow** [comment]
You can also create a MkDocs Hook which replaces any `https://cdn.jsdelivr.net/gh/jdecked/twemoji@14.1.2/` prefix with your own path to the previously downloaded SVG assets directory.

But paying for Insiders and using the privacy plugin would be easier and faster to do ✌️

**Reply by squidfunk** [reply]
Thanks for sharing – jup, anything is possible to achieve with customization. In the end, the question is how much code you want to maintain yourself and how much time you want to put into a solution we've already built and that is battle-tested ☺️ Also, if you run into any bugs with the privacy plugin, we're more than happy to fix them as quickly as we can 🚀
